### ISSUE TITLE
[Bug]: Increment pub date by minutes instead of seconds for audiobook tracks

### ISSUE THREAD

**Issue by TOZigs** [advplyr/audiobookshelf#3442]
### What happened?

I'm trying to add my audiobooks to an rss feed to add to my podcatcher.
I'm doing this by making an rss feed to a collection. 
But it seems like the publication date of my "episodes" (read chapters) change based on the book I first added to my library.

What happens is that all my books are mixed up with the earliest chapters coming in like: book 1 chapter 1, book 2 chapter 1, book 3 chapter 1, book 1 chapter 2 etc

### What did you expect to happen?

I expected that my episodes would be sorted by book in the right order of chapters. 

### Steps to reproduce the issue

1. Add one book to a collection
2. Pull the data from the rss feed 

The result is as seen in the following screenshots:
![Screenshot_20240922_073115_Chrome](https://github.com/user-attachments/assets/c97172da-7cc4-4ed1-88ad-3c4f43cc1cee)

![Screenshot_20240922_073223_Chrome](https://github.com/user-attachments/assets/ce003837-f936-41c7-8e7c-3309f2fff967)

### Audiobookshelf version

2.13.4

### How are you running audiobookshelf?

Docker

### What OS is your Audiobookshelf server hosted from?

Windows

### If the issue is being seen in the UI, what browsers are you seeing the problem on?

None

**Comment by advplyr** [comment]
I'm not able to reproduce the issue but I'm not sure I understand. The pub date is somewhat arbitrary for audiobooks because all that matters is that it is in the correct order.
The order will be the order the books are in your collection. Then each books chapters/episodes will be ordered based on the tracks of that book.
If you have episodes that are out of order then make sure that the tracks on the book are in the correct order. You can manage the track order with the "Manage Tracks" button 
![image](https://github.com/user-attachments/assets/a1a27570-9512-446c-8d78-babc2a8fd85e)

**Comment by hobesman** [comment]
This just happened to me, using Pocket Casts. With multiple books in a collection shared via RSS, it seems like maybe Pocket Casts is ignoring the publication "seconds" and only reading the "day" and therefore acting like they're all published the same time. It then falls back to sorting alphabetically for episodes published "simultaneously."

![Screenshot_20250115_093925_Pocket Casts](https://github.com/user-attachments/assets/cae82c28-391b-41b0-b162-dc355ee27d55)

This screenshot is with the Pocket Casts app set to sort episodes oldest to newest.

**Comment by advplyr** [comment]
When you open the RSS feed in the browser you can see that the episodes are in order there?

**Comment by hobesman** [comment]
> When you open the RSS feed in the browser you can see that the episodes are in order there?

Yes, confirmed. The books are each 4 tracks, and in the browser the RSS URL shows shows track 1, 2, 3, 4, 1, 2, 3, 4, 1, 2...

**Comment by advplyr** [comment]
Since I don't have pocket casts one thing you can do that would be helpful to test is to add a podcast in a podcast library and edit the pub dates manually before opening an RSS feed.

![Image](https://github.com/user-attachments/assets/faebd29a-b833-41dd-8e7a-252f5d297501)

If it is the case that we need to space out the pub dates for audiobook tracks you should be able to find out how much we need to do

**Comment by hobesman** [comment]
Ok I reshared that new podcast from ABS and added to Pocket Casts, and that app honored the episodes being published different minutes, but ignored differences in seconds for publishing date.

**Comment by advplyr** [comment]
I don't think we're on the same page. 

From my understanding of your original issue the way that Abs is setting the pub date for audiobooks (by incrementing each tracks pub date by 1 second) is not working in Pocket Casts.

Your guess is that the reason it isn't working is because the 1 second increment is not large enough for Pocket Casts because Pocket Casts may be truncating by day. My guess would be that is unlikely but they may be truncating by minute or hour.

My suggestion as a way to test if this is to add a new podcast just for testing. You can use the "Add" button and search itunes to add a random podcast is probably the easiest way. I'll outline this step by step

1. Add a new podcast with some episodes
2. Navigate to the podcast page and edit the Pub Date field for each episode (my screenshot above).
   * The pub dates you can space out by minutes, hours, or day for testing to see which one is detected.
3. Open an RSS feed for this podcast (screenshot below)
4. Put the Abs RSS feed URL into Pocket Casts

![Image](https://github.com/user-attachments/assets/6af3b7ef-6e69-44b1-b345-577800adcac6)

**Comment by hobesman** [comment]
Yes that's essentially what I tested. I imported the podcast I'd shared. I changed some episodes by minutes, some by days, and left some unchanged (ABS doesn't let me see or change publication seconds). I then shared that podcast and added to Pocket Casts. The app honored the days and minutes differences and sorted accordingly, but ignored differences of seconds and lumped those all as simultaneous (i.e. reverted to alphabetical sorting)

**Comment by advplyr** [comment]
Yeah we can do that. I'm not sure that this issue is the same but I suppose we could use it

**Comment by hobesman** [comment]
Thanks for the update. I tested this today and it appears to increment by 1 minute as intended, BUT it resets the starting publish time for each book, so sorting by date is still problematic. Any idea why the publish time for each resets for each book instead of continuing from the book before in the connection?

**Comment by advplyr** [comment]
Ah yeah I know why

**Comment by hobesman** [comment]
That seems to have done the trick. Thanks!

Note: the Pocket Casts app seems to have cached the publication times. I hadn't actually added the podcast to the PC app, only searched and previewed it. After this update I closed the feed and reopened it, viewed the html to confirm the correct increment was being applied, and then went to search it again in the PC app, only to see the episodes still in the wrong order. 

So then, back in ABS, I closed the feed, and reopened with a different slug. Searching the new feed in the PC app showed the tracks in the correct order. So ABS was using the correct increment and publication timing, but PC was still showing me the old info, even though I never officially "added" the feed in PC, only searched and previewed it. Just some extra data in case future feed issues arise when interfacing with other apps.
